### PR TITLE
Fix #283964 - Linked parts for single voices don't work

### PR DESCRIPTION
**Pull request by njvdberg**
Resolves: https://musescore.org/en/node/283964

Corrects recognition of linked staves with a part and linked staves over
excerpts.

<code>Score::undoAddCR()</code> will find all linked staves for the passed <code>ChordRest</code> and iterates over all these staves and decides what to do. The root cause of this issue was this algorithm didn't clearly distinguish between the two different kind of linked staves:
- 1 Linked staves within a part
- 2 Linked staves between master score and scores on excerpt.

This PR now handles creating a <code>ChordRest</code> for the following cases:
- 1 On non-linked staff at the master score.
- 2 On a staff linked, via an excerpt, to another score on another part.
- 3 On a staff linked, via an excerpt, to another score on another part using the voice-to-part feature.
- 4 On a staff linked to another staff on the same score ("linked staves" in the <code>Parts<code> dialog).
- 5 On a staff linked to another staff on the same score **and** to another part.
- 6 On a staff linked to another staff on the same score **and** to another part using the voice-to-part.

All 6 cases are tested by creating a <cod>ChordRest</code> on the master score and all, if applicable, on any of the linked staves. 

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by njvdberg** [comment]
Travic CI failure pending the merge of PR #6023

**Comment by Jojo-Schmitz** [comment]
#6023 has been merged, so I guess you'd need to rebase to get past that Travis error?

File: libmscore/edit.cpp [hunk 1]
```diff
@@ -4878,17 +4878,46 @@ void Score::undoAddCR(ChordRest* cr, Measure* measure, const Fraction& tick)
 
       Tuplet* t = cr->tuplet();
 
+      // For linked staves the length of staffList is always > 1 since the list contains the staff self too!
```
Comment (Jojo-Schmitz) [inline libmscore/edit.cpp:4881]: self -> itself?

File: libmscore/edit.cpp [hunk 2]
```diff
@@ -4878,17 +4878,46 @@ void Score::undoAddCR(ChordRest* cr, Measure* measure, const Fraction& tick)
...
+      const bool linked = ostaff->staffList().length() > 1;
+
       for (const Staff* staff : ostaff->staffList()) {
+
+
```
Comment (Jojo-Schmitz) [inline libmscore/edit.cpp:4886]: excessive whitespace

**Comment by Jojo-Schmitz** [comment]
Hmm, Not good, may need to rebase from 3.x I guess

File: libmscore/edit.cpp [hunk 3]
```diff
@@ -4878,17 +4878,45 @@ void Score::undoAddCR(ChordRest* cr, Measure* measure, const Fraction& tick)
 
       Tuplet* t = cr->tuplet();
 
+      // For linked staves the length of staffList is always > 1 since the list contains the staff itself too!
+      const bool linked = ostaff->staffList().length() > 1;
+
       for (const Staff* staff : ostaff->staffList()) {
+
```
Comment (Jojo-Schmitz) [inline libmscore/edit.cpp:4885]: Still an empty line too much IMHO

**Comment by njvdberg** [comment]
Wrong branch indeed, still confusing :-(

File: libmscore/edit.cpp (hunk 3 above)
Comment (njvdberg) [reply libmscore/edit.cpp:4885]: Removed only one line. Now the other is gone too.

**Comment by Jojo-Schmitz** [comment]
Don't forget to update https://github.com/musescore/MuseScore/issues/6027#issuecomment-623123228 ;-)

**Comment by njvdberg** [comment]
I didn't, was done already ;-)

**Comment by Jojo-Schmitz** [comment]
No, this PR is still mentioned there

**Pull request closed without merging**